src/Mjolnir.ts

```diff
@@ -375,72 +335,91 @@ export class Mjolnir {
+    /**
+     * Protect the room, but do not persist it to the account data.
+     * @param roomId The room to protect.
+     */
+    private protectRoom(roomId: string): void {
+        this.protectedRoomsTracker.addProtectedRoom(roomId);
+        this.roomJoins.addRoom(roomId);
     }
 
+    /**
+     * Remove a room from the explicitly protect set of rooms that is persisted to account data.
+     * Should NOT be used to remove a room that we have left, e.g. when implementing `config.protectAllJoinedRooms`,
+     * use `unprotectRoom` instead.
+     * @param roomId The room to remove from account data and stop protecting.
+     */
     public async removeProtectedRoom(roomId: string) {
-        delete this.protectedRooms[roomId];
+        await this.protectedRoomsConfig.removeProtectedRoom(roomId);
+        this.unprotectRoom(roomId);
+    }
+
+    /**
+     * Unprotect a room.
+     * @param roomId The room to stop protecting.
+     */
+    private unprotectRoom(roomId: string): void {
         this.roomJoins.removeRoom(roomId);
         this.protectedRoomsTracker.removeProtectedRoom(roomId);
-
-        const idx = this.explicitlyProtectedRoomIds.indexOf(roomId);
-        if (idx >= 0) this.explicitlyProtectedRoomIds.splice(idx, 1);
-
-        let additionalProtectedRooms: { rooms?: string[] } | null = null;
-        try {
-            additionalProtectedRooms = await this.client.getAccountData(PROTECTED_ROOMS_EVENT_TYPE);
-        } catch (e) {
-            LogService.warn("Mjolnir", extractRequestError(e));
-        }
-        additionalProtectedRooms = { rooms: additionalProtectedRooms?.rooms?.filter(r => r !== roomId) ?? [] };
-        await this.client.setAccountData(PROTECTED_ROOMS_EVENT_TYPE, additionalProtectedRooms);
     }
 
-    // See https://github.com/matrix-org/mjolnir/issues/370.
-    private async resyncJoinedRooms(withSync = true) {
+    /**
+     * Resynchronize the protected rooms with rooms that the mjolnir user is joined to.
+     * This is to implement `config.protectAllJoinedRooms` functionality.
+     * @param withSync Whether to synchronize all protected rooms with the watched policy lists afterwards.
+     */
+    private async resyncJoinedRooms(withSync = true): Promise<void> {
         if (!this.config.protectAllJoinedRooms) return;
 
-        const joinedRoomIds = (await this.client.getJoinedRooms())
-            .filter(r => r !== this.managementRoomId && !this.unprotectedWatchedListRooms.includes(r));
-        const oldRoomIdsSet = new Set(this.protectedJoinedRoomIds);
-        const joinedRoomIdsSet = new Set(joinedRoomIds);
+        // We filter out all policy rooms so that we only protect ones that are
+        // explicitly protected, so that we don't try to protect lists that we are just watching.
+        const filterOutManagementAndPolicyRooms = (roomId: string) => {
+            const policyListIds = this.policyLists.map(list => list.roomId);
```

Comment:
Literally: because `includes` doesn't accept a key function where i can put `list.roomId` instead.

Really: I just find it much simpler than tracking yet another group/category of rooms. I don't think it is costing us much to do this either.